fe/fe-core/src/main/java/com/starrocks/sql/optimizer/cost/feature/OperatorFeatures.java

```diff
@@ -155,6 +168,68 @@ public Table getTable() {
             return table;
         }
 
+        public TableFeature getTableFeature() {
+            return new TableFeature(table, statistics);
+        }
+    }
+
+    /**
+     * Table feature, used to extract table features from scan operators
+     */
+    public static class TableFeature {
+        private final Table table;
+        private final Statistics statistics;
+
+        public TableFeature(Table table, Statistics statistics) {
+            this.table = table;
+            this.statistics = statistics;
+        }
+
+        public Table getTable() {
+            return table;
+        }
+
+        public long getRowCount() {
+            if (table instanceof OlapTable) {
+                return ((OlapTable) table).getRowCount();
+            } else {
+                return (long) statistics.getOutputRowCount();
+            }
+        }
+
+        /**
+         * Identifier of the table, used to identify the table in the feature vector
+         */
+        public String getTableIdentifier() {
+            if (table instanceof OlapTable) {
+                return String.valueOf(((OlapTable) table).getId());
+            } else {
+                try {
+                    String dbName = table.getCatalogDBName();
```

Comment:
we only implemented `getTableIdentifier` on Hive/Iceberg/Hudi/deltaLake tables

Reply:
`TableFeature` is only one of all features to mark tables used in query which only takes 3 tables.

For un-implemented tables, use `table.getName()` instead which I think it's enough in most cases.